### DISCUSSION TITLE
OpenAPI endpoint and SwaggerUI or Redoc or other clients

### DISCUSSION THREAD

**Discussion by alessandromarchetti** [medusajs/medusa#3671] [Q&A] [answered]
Hi, I have 2 questions about medusa server and I couldn't find the information in the docs:

- Is there one or more endpoints exposing auto-generated OpenAPI specs for the APIs (both store and admin and, possibly, custom endpoints added via customization)?
- If the answer is yes, is there a SwaggerUI or a Redoc endpoint (or any other web client) which renders the documentation?

Thanks in advance for anyone who could help
Alessandro

**Comment by srindom** [comment, marked as answer]
Hi @alessandromarchetti! We don't have an endpoint on the server that exposes this, but @patrick-medusajs has recently built a powerful OpenAPI Spec generator which will be released in v1.8. You can check out the release candidate for more details: https://github.com/medusajs/medusa/releases/tag/v1.8.0-rc.0

**Reply by alessandromarchetti** [reply]
Thank you!

**Reply by alessandromarchetti** [reply]
Does the OpenAPI Spec generator also include custom endpoints, whether they are added or overridden via standard medusa plugins or by using medusa-extender?

**Reply by patrick-medusajs** [reply]
Yes it will. By passing the directory path to codebase containing the custom endpoints, the OAS generator will crawl and discover [swagger-inline](https://github.com/readmeio/swagger-inline) OAS comments. Any paths or schemas discovered will be combined into Medusa's default OAS. The package's [readme](https://github.com/medusajs/medusa/tree/v1.8.0-rc.0/packages/oas/medusa-oas-cli) can help you get you started with the new `medusa-oas` CLI.

**Reply by NicolasGorga** [reply]
Hey @patrick-medusajs i am having trouble generating the oas for customizations found in my project. I am passing the directory to scan via --paths, but when debugging in node_modules, i see at some point a swagger-inline dependency, incorrectly generates a Regex to match against the discovered files in the directory for the generator to use as input. 

I am on Windows, and the positive regex looks like: /^(?:E:\proyectos personales\mercur\apps\backend\src\api(?:[\\/](?!\.)(?:(?:(?!(?:^|[\\/])\.).)*?)|$))$/ the problem i see is that '\' are not escaped correctly (should be '\\'). Do you have any input on this?
